util/imageutil/config.go

```diff
@@ -213,10 +214,25 @@ func childrenConfigHandler(provider content.Provider, platform platforms.MatchCo
 			}
 
 			if platform != nil {
-				for _, d := range index.Manifests {
-					if d.Platform == nil || platform.Match(*d.Platform) {
-						descs = append(descs, d)
-					}
+				if len(index.Manifests) > 0 {
+					// Here we want to select only the best manifest for the platform.
+					// We do not want every platform or even every matching platform.
+					// Whatever is selected here will, depending on resolve mode, be checked for locally and then fetched if it is not present.
+					// Selecting only the best match means we are making fewer fewer round trips with the content store and the registry.
+					// It also means we may be able to avoid talking to the registry at all.
+					// This is important for cases where the image doesn't actually exist in the registry (yet) but we have a local copy.
+					sort.SliceStable(index.Manifests, func(i, j int) bool {
+						mi := index.Manifests[i]
+						if mi.Platform == nil {
+							return false
+						}
+						mj := index.Manifests[j]
+						if mj.Platform == nil {
+							return true
+						}
+						return platform.Less(*mi.Platform, *mj.Platform)
+					})
+					descs = append(descs, index.Manifests[0])
```

Comment:
Shouldn't we ALSO keep the previous check for `manifest.Platform == nil || platform.Match(*manifest.Platform)` ?

Otherwise in a case where the requested platform doesn't match any manifest, it would just return whatever happen to be first in the manifest index (which theoretically, could even be something like an attestation manifest with platform `unknown/unknown`).

Reply:
+1 a test for this case would be nice too, just in case

Reply:
Fixed.